### DISCUSSION TITLE
Single Select Field 25 Maximum Number of Options is Too Low

### DISCUSSION THREAD

**Discussion by ynouri** [community/community#6419] [Projects and Issues]
We use the Single Select field to group our issues by epics. We have way more than 25 epics, and this is blocking us to create new epics without removing others. The maximum should be much higher (arbitrary suggestion: maybe 1,000 options)?

![image](https://user-images.githubusercontent.com/1789921/137596449-65c37775-18e6-4930-8842-4293dcff0317.png)

**Comment by Totktonada** [comment]
We (in @tarantool) have exactly same case with epics and exactly same problem (described in https://github.com/github/feedback/discussions/6525).

**Comment by soywiz** [comment]
In my projects I have a single selection option to describe components, and reached the 25 limit already:
https://github.com/orgs/korlibs/projects/19/views/1

50 might be enough, but maybe 100 would be also nice?

**Comment by despairblue** [comment]
Just a thought, but aren't milestones the equivalent to epics?
Otherwise you can use a number (or text, if you need the padding with zeros for sorting). 
I'm using that for cycles (pretty much just long sprints). You can still group and filter by them. 

I don't think a single selection field is a good fit for an infinite resource like sprints or epics.

The components use case though seem genuine. Could be worked around with labels maybe though.

**Reply by ynouri** [reply]
Right now we are using Milestones to represent a sprint. We are thinking about moving to the new Cycle functionality but it still lacks some flexibility preventing us to adopt (e.g. fully custom dates) Also it's not clear to me whether it is the best practice to represent epics through Milestones? Is there some official documentation about this?

Re single selection with high cardinality, we are doing this in Notion by linking 2 databases (one for epics and one for issues) and it works wonderfully. I don't see why it couldn't work in Github. One key feature currently missing though is a good search mechanism when associating an issue to a single selection option (otherwise you'd have to scroll through hundred of options)

**Comment by rafal-myszynski** [comment]
I agree, 25 is too low, I reached today. For me, in that case, 100 would be good.
I'm using Single Select to assign the Area of tasks, similar like is presented in New Issues promo materials on https://github.com/features/issues

![issues-area](https://user-images.githubusercontent.com/31932992/141614228-97cc558a-96f6-4ab3-8a02-50f43d3b5bce.png)

**Reply by stuntbox** [reply]
Huge +1 to this. 25 is far too low and we hit our head on it almost immediately.

**Comment by borjasanlei** [comment]
I'm using Single Select options for linking the issue/card with the internal Id of the project/module developed, for example `awesome_project/module1`, `awesome_project/module2` or `nice_web/module1`.

25 limit is a bottleneck for us, since:

- Creating individual Github projects for each dev project is not an option
- Not all cards are converted to issues, thus replacing them with labels is neither an option

:+1: to this question!

**Comment by amadeogallardo** [comment]
We're using this to track which customer(s) requested/reported the issue, and the 25 maximum options doesn't really work out for us.

**Reply by amadeogallardo** [reply]
Seems like the limit has been updated to 50 now.

**Comment by dhpizza** [comment]
Still, why only 50? Would be nice to have a couple 💯 🙏

**Comment by boblenclume** [comment]
Would it be possible to upgrade the limit to 150 or 200 ? 🙏
Using it to map some fields allowing to tag my cards with. But getting :
`     
  "message" => "Settings options cannot contain more than 100 entries"
`

**Comment by Alex-ley-scrub** [comment]
Can we put the limit to 1,000 or more please so we don't have to ask for it to be increased every year 🤣

**Reply by Alex-ley-scrub** [reply]
OR can we make it limitless? Like labels?
If this is not technically feasible for some reason, can you also make it easier to convert a single select field into a text field so we can get around this issue in a simpler way (compared to writing a script to do this etc.)

**Reply by Alex-ley-scrub** [reply]
AND maybe warn people of the 100 limit up front when they create the new field instead of 6-18 months down the line when they realise too late?

**Reply by iansimpson-st** [reply]
Seriously.  Why is there a limit at all?
